src/IW/Sync/Search.hs

```diff
@@ -28,70 +32,109 @@ import qualified Data.Text as T
 import qualified Data.Vector as V
 
 
--- | Fetch all repositories built with the Haskell language.
-fetchAllHaskellRepos
-    :: ( MonadIO m
-       , WithError m
-       , WithLog env m
-       )
-    => m [GitHub.Repo]
-fetchAllHaskellRepos = liftGitHubSearchToApp searchHaskellRepos
-
--- | Convert a value of the @GitHub.Repo@ type to a value of our own @Repo@ type.
-fromGitHubRepo :: GitHub.Repo -> Repo
-fromGitHubRepo githubRepo = Repo
-    { repoOwner      = RepoOwner $ GitHub.untagName $ GitHub.simpleOwnerLogin $ GitHub.repoOwner githubRepo
-    , repoName       = RepoName  $ GitHub.untagName $ GitHub.repoName githubRepo
-    , repoDescr      = fromMaybe "" $ GitHub.repoDescription githubRepo
-    , repoCategories = SqlArray []
-    }
-
--- | Fetch all open issues with Haskell language and the labels passed in to the function.
-fetchHaskellIssuesByLabels
-    :: ( MonadIO m
-       , WithError m
-       , WithLog env m
-       )
-    => [Label]
-    -> m [GitHub.Issue]
-fetchHaskellIssuesByLabels = liftGitHubSearchToApp . searchHaskellIssuesByLabels
-
--- | Fetch all open issues with Haskell language.
-fetchAllHaskellIssues
-    :: ( MonadIO m
-       , WithError m
-       , WithLog env m
-       )
-    => m [GitHub.Issue]
-fetchAllHaskellIssues = fetchHaskellIssuesByLabels []
-
--- | Lift a github search action to work within our monad stack.
-liftGitHubSearchToApp
+liftGithubSearchToApp
     :: forall a env m.
        ( MonadIO m
+       , MonadUnliftIO m
        , WithError m
        , WithLog env m
        , Typeable a
        )
     => IO (Either GitHub.Error (SearchResult a))
     -> m [a]
-liftGitHubSearchToApp githubSearch = liftIO githubSearch >>= \case
+liftGithubSearchToApp githubSearch' = do
```

Comment:
Can we add comments to this logic? It's quite complicated and future us would be thankful for that :slightly_smiling_face: